up_fast_downward/fast_downward.py

```diff
@@ -257,3 +259,82 @@ def supports(problem_kind: "ProblemKind") -> bool:
     @staticmethod
     def satisfies(optimality_guarantee: OptimalityGuarantee) -> bool:
         return True
+
+    # To avoid the introduction of axioms with complicated goals, we introduce
```

Comment:
@roeger Does this add any overhead?
If yes, maybe you can avoid the introduction of this extra action when complicated goals are not present.

Reply:
The overhead is negligible. I will still consider to only apply it only when necessary for the grounder because there I don't know what will happen to the task afterwards.